libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,83 @@
+/* OTA task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include "HttpsOTAUpdate.h"
+#include "Esp.h"
+#define OTA_TASK_STACK_SIZE 9216
+
+static esp_http_client_config_t config; 
+static void (*http_event_cb)(HttpEvent_t*);
+
+static EventGroupHandle_t ota_status;//check for ota status
+const int OTA_SUCCESSFUL_BIT = BIT0;
+const int OTA_FAILURE_BIT = BIT1;
+
+static int status = -1;
+
+esp_err_t http_event_handler(esp_http_client_event_t *event)
+{
+    http_event_cb(event); 
+    return ESP_OK;
+}
+
+void https_ota_task(void *param)
+{
+    esp_err_t ret = esp_https_ota((const esp_http_client_config_t *)param);
+    if(ret == ESP_OK) {
+        xEventGroupSetBits(ota_status, OTA_SUCCESSFUL_BIT);
+    } else {
+        xEventGroupSetBits(ota_status, OTA_FAILURE_BIT);
+        vTaskDelete(NULL);
+    }
+    while (1) {
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+
+int HttpsOTAUpdateClass::otaStatus()
```

Comment:
Why do you need ota in the name? The class has the name, so you can rename this to status?